SPHINXsys/src/shared/particle_dynamics/solid_dynamics/contact_dynamics.cpp

```diff
@@ -59,6 +59,64 @@ namespace SPH
 			contact_density_[index_i] = sigma;
 		}
 		//=================================================================================================//
+		ShellContactDensitySummation::
+			ShellContactDensitySummation(SolidBodyRelationContact *solid_body_contact_relation)
+			: PartInteractionDynamicsByParticle(solid_body_contact_relation->sph_body_,
+												&solid_body_contact_relation->body_surface_layer_),
+			  ContactDynamicsData(solid_body_contact_relation), pos_n_(particles_->pos_n_),
+			  contact_density_(particles_->contact_density_), n_0_(particles_->n_0_),
+			  kernel_(solid_body_contact_relation->sph_body_->particle_adaptation_->getKernel()), 
+			  spacing_ref_(solid_body_contact_relation->sph_body_->particle_adaptation_->ReferenceSpacing())
+		{
+			for (size_t k = 0; k != contact_particles_.size(); ++k)
```

Comment:
can we use < instead? I think this can cause bugs

Reply:
this issue was raised before by me and I and @Xiangyu-Hu concluded this is not an issue. This might have some very minor positive effects on performance. Some stackoverflowers suggested to do this.